m01_introduction/01_gallery.rst

```diff
@@ -0,0 +1,158 @@
+
+.. _front-page:
+
+Front page
+----------
+
+.. figure:: /images/gallery/aplus-front-page.png
+  :width: 80%
+  :align: center
+
+  **A+ front page Spring-2020.** In the A+ front page you can find the list of the courses (ongoing courses) uploaded in the platform`.
+
+The above image shows a screenshot of the front page of `A+ at Aalto University
+<https://plus.cs.aalto.fi/>`_ from Spring 2020. The front page shows the most recent courses. Each course has an optional logo image, name, course
+code and teaching time. All courses have a green "Aalto" tag to indicate that
+the course is for Aalto University students. The Ohjelmointistudio 2 course
+has also a blue "MOOC" tag to indicate the course is also a massive open online
+course. Sections :doc:`9.1 Settings a course on production servers
+</m09_admin/01_setup>`
+and :doc:`9.2 Course settings </m09_admin/02_settings>` describe these in detail.
+
+Exercises
+---------
+
+Point and click
+...............
+
+.. figure:: /images/gallery/jsvee-factorial.png
+  :width: 80%
+  :align: center
+
+  A **JSVEE exercise** is a slideshow of program execution. 
+
+This particular example is from the *Data Structures and Algorithms Y* course, which shows how a factorial function in Python is executed.
+
+.. figure:: /images/gallery/jsav-minheap-insert.png
+  :width: 80%
+  :align: center
+
+  A **JSAV exercise** is a visual algorithm simulation exercise.
+
+The student manipulates a data structure with given input data according to instructions
+   by pointing and clicking with the mouse. This particular example is about
+   inserting elements to a minimum heap.
+
+   After the student has worked with the exercise, they can submit it and
+   receive points automatically. They also can review a model example or reset
+   the exercise with new initial data. These exercises are used on the
+   *Data Structures and Algorithms* courses (both *Y* and *SCI* versions for
+   students of Computer Science minor / major).
+
+
+Databases
+.........
+
+.. figure:: /images/gallery/relational-algebra.png
+  :width: 80%
+  :align: center
+
+  A **relational algebra exercise** on the *Databases* course. The student
+  constructs a database query in relational algebra, which is then
+  automatically graded.
+
+Python
+......
+
+.. figure:: /images/gallery/python-fractal-graded.png
+
+   A **Programming exercise with graphical output**. This is a hard exercise
+   from the course *Data Structures and Algorithms Y*, where the student
+   writes a Python program that draws a bitmap fractal similar to
+   `Sierpinski carpet <https://en.wikipedia.org/wiki/Sierpinski_carpet>`_.
+   The picture shows the feedback for the student from the automatic grader.
+   There are Python unit tests which give points, and also a visual comparison
+   of the outputs of the student's program and the correct solution.
+
+:doc:`Section 4 </m04_programming_exercises/01_instructions/>`
+describes how to create Python programming exercises.
+
```

Comment:
Should we add some examples of Acos exercises here? The gallery should not leave the impression that only these few example exercise types are possible to implement in A+. The introduction of this chapter could also emphasize that the implementation of exercises is very flexible, but this page lists some examples.